docs/_tutorials/read-only-json-api.md

```diff
@@ -0,0 +1,416 @@
+ --- you will generate a custom one later. Generating pages like `about` is out of scope for this tutorial, but in case
+you need them, they behave the same way that posts do.
+
+### Dummy Posts
+
+Under `_posts`, create two new documents. Remember that Jekyll requires you to **prepend post filenames with a date**:
+
+`2020–12–20-the-first-post.md`
+
+```md
+---
+title:  "The First Post"
+category: update
+custom-property: "My custom property"
+---
+1
+```
+
+Make sure to include a `custom-property` so that you can fetch it later on. 
+
+`2020–12–20-the-second-post.md`
+
+```md
+---
+title:  "The Second Post"
+categories: ["update", "tutorial"]
+---
+2
+```
+
+This one belongs to two categories --- `update` and `tutorial` --- and doesn't include the custom property. Because the
+categories overlap, it will be possible to test getting a list of posts from a category.
+
+### Layouts
+
+The last thing is to define the layouts. First, declare them in `config.yml`:
+
+```yml
+defaults:
+  - scope:
+      type: categories
+    values:
+      layout: items_index
+      permalink: categories/:category/
+  - scope:
+      type: categories_index
+    values:
+      layout: categories_index
+      permalink: categories/
+  - scope:
+      type: posts
+    values:
+      layout: post
+  - scope:
+      type: posts_index
+    values:
+      layout: items_index
+```
+
+* You have four various content types: an index of all categories, a listing of all posts under a category, a 
+listing of all posts in the system and a single post. 
+* You specify a layout using a `layout` key.
+* You will reference `type` when you'll start writing the plugin. For purposes of this tutorial, it's only 
+required to be unique.
+* For some of the items you declare permalink. You can do it for all of them. It can come in handy if you want to 
+define your own API endpoints.
+
+Now you can define them. Before you do it, it's worth noting that you can list all the available properties with this little template, which
+returns an array of available keys:
+
+{% raw %}
+```liquid
+{{ page.keys | jsonify }}
+```
+{% endraw %}
+
+If you enter it in a `post.html` template and execute the following command:
+
+```sh
+bundle exec jekyll build
+```
+
+You'll get this in `/_site/update/2020/12/20/the-first-post.html`:
+
+```json
+["url","relative_path","excerpt","title","categories","path","next",
+ "previous","id","output","date","tags","content","collection",
+ "draft","layout","category","custom-property","slug","ext"]
+```
+
+Given that, I'd like you to enter the following template:
+
+{% raw %}
+```liquid
+{% include post.html post=page %}
+```
+{% endraw %}
+
+* Include a partial template `post.html` from `_includes` directory. 
+* Assign a value to internal `post` variable
+
+Partial template `post.html` should look like this:
+
+{% raw %}
+```liquid
+{
+    "title": {{ include.post.title | jsonify }},
```

Comment:
what happens if one were to use `"title": {{ page.title | jsonify }}` instead?
`

Reply:
A listing of posts under a category would render with `null` values.